### PR TITLE
Prevents Midrounds From Spawning After Shuttlecall

### PR DESCRIPTION
**Pull request by eeSPee**
## About The Pull Request

Prevents certain events from spawning after shuttle has been called.

## Why It's Good For The Game

It's not fun when you're midround alien/abductor/pirate and you have less than 10 minutes to do your thing.

## Changelog
:cl:
tweak: Prevents midround antags from spawning after shuttle has been called.
/:cl:

### CODE REVIEW

File: code/modules/events/_event.dm [hunk 1]
```diff
@@ -30,8 +30,9 @@
 	var/triggering	//admin cancellation
 	var/auto_add = TRUE				//Auto add to the event pool, if not you have to do it yourself!
 
-	/// Whether or not dynamic should hijack this event
-	var/dynamic_should_hijack = FALSE
+	
+	var/dynamic_should_hijack = FALSE	// Whether or not dynamic should hijack this event	
+	var/can_spawn_after_shuttlecall = FALSE	// Prevents the event from spawning after the shuttle was called
```
Comment (PowerfulBacon) [inline code/modules/events/_event.dm]: Why is the name of the variable the literal opposite of what it does. This doesn't allow them to spawn after shuttlecall, this prevents them from spawning after shuttle call if true.

File: code/modules/events/blob.dm [hunk 2]
```diff
@@ -7,6 +7,7 @@
 	min_players = 20
 
 	dynamic_should_hijack = TRUE
+	can_spawn_after_shuttlecall = TRUE
```
Comment (PowerfulBacon) [inline code/modules/events/blob.dm]: Blob delays shuttle anyway

**Review by PowerfulBacon — DISMISSED** [review]

**Comment by Sinestia** [comment]
I'm gonna powergame this by keeping the shuttle called as much as possible /s


Aliens delay the shuttle too if they get an empress/queen up in time, it might be worth allowing them to spawn too. 

Up to you though, probably isn't as warranted as it is with blob

**Comment by OceanFish1** [comment]
Go on green alert, call shuttle, recall in 9 minutes, repeat,????, no more midround antags

**Comment by wraith-54321** [comment]
there is a thing if you keep calling and recalling it forces an un-recallable shuttle call, so I dont think that would end up happening unless people knew about this but not that

**Comment by Sinestia** [comment]
It doesn't really matter

 If you do that it's the easiest thing to ban for + apparently there actually are safeguards to ping pong as mentioned if you go hard enough.   

 Usually I'm more along the lines of "jannies suck and its best thing things are addressed by code" but this is sooooo blatant if you actually are dumb enough to do it.

**Comment by ike709** [comment]
inb4 people spam the shuttle to reduce antags

**Comment by PowerfulBacon** [comment]
> 
> 
> inb4 people spam the shuttle to reduce antags

I mean that's likely not going to be a huge issue and is pretty obvious for admins to deal with, you could literally go to their spawn points and wall them off if you wanted to meta that hard.

**Review by ike709 — APPROVED** [review]

**Comment by eeSPee** [comment]
> > inb4 people spam the shuttle to reduce antags
> 
> I mean that's likely not going to be a huge issue and is pretty obvious for admins to deal with, you could literally go to their spawn points and wall them off if you wanted to meta that hard.

Assistant McAssistantson walling off gateway and welding off all vents outside main hallways :S

**Pull request merged**